src/main.rs

```diff
@@ -232,11 +300,214 @@ mod cli {
             Kanata::start_notification_loop(nrx, server.connections);
         }
 
+        // Start file watcher if enabled
+        #[cfg(feature = "watch")]
+        if args.watch {
+            if let Err(e) = file_watcher::start_file_watcher(args.paths.clone(), kanata_arc.clone()) {
+                log::error!("Failed to start file watcher: {}", e);
+            }
+        }
+
         #[cfg(target_os = "linux")]
         sd_notify::notify(true, &[sd_notify::NotifyState::Ready])?;
 
         Kanata::event_loop(kanata_arc, tx)
     }
+
+    #[cfg(all(feature = "watch", test))]
+    mod tests {
+        use super::*;
+        use std::fs;
+        use tempfile::NamedTempFile;
+
+        #[test]
+        fn test_request_live_reload_method() {
+            // Test that the new request_live_reload method works
+            let temp_file = NamedTempFile::with_suffix(".kbd").unwrap();
+            let config_path = temp_file.path().to_path_buf();
+            
+            // Write a minimal valid config
+            fs::write(&config_path, r#"
+                (defsrc caps)
+                (deflayer default caps)
+            "#).unwrap();
+
+            let args = ValidatedArgs {
+                paths: vec![config_path],
+                #[cfg(feature = "tcp_server")]
+                tcp_server_address: None,
+                #[cfg(target_os = "linux")]
+                symlink_path: None,
+                nodelay: true,
+                #[cfg(feature = "watch")]
+                watch: false,
+            };
+
+            let mut kanata = Kanata::new(&args).unwrap();
+            
+            // Initially, live_reload_requested should be false
+            assert!(!kanata.is_live_reload_requested());
+            
+            // Request live reload
+            kanata.request_live_reload();
+            
+            // Now it should be true
+            assert!(kanata.is_live_reload_requested());
+        }
+
+        #[test]
+        fn test_file_watcher_path_validation() {
+            // Test that file watcher handles various path scenarios correctly
+            let valid_paths = vec![
+                PathBuf::from("/tmp/test.cfg"),
+                PathBuf::from("./relative.cfg"),
+                PathBuf::from("/home/user/config.kbd"),
+            ];
+            
+            // All paths should be valid for direct file watching
+            for path in &valid_paths {
+                assert!(path.as_os_str().len() > 0, "Config path should not be empty: {}", path.display());
+                assert!(path.extension().is_some(), "Config file should have an extension: {}", path.display());
+            }
+        }
+
+        #[test]
+        fn test_validated_args_with_watch_flag() {
```

Comment:
This doesn't appear to test anything other than the test code itself. I guess in other contexts an initializer or set/get property could have obfuscated behaviour, but this test code has zero possibility of that. This test should be removed.